evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -707,39 +710,132 @@ def test_component_graph_dataset_with_different_types():
+                                             text_columns)
+    assert input_feature_names['Random Forest'] == (['column_3', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
+                                                     'column_2_1', 'column_2_2', 'column_2_3', 'column_2_4', 'column_2_5', 'column_2_6',
+                                                     'column_4_year', 'column_4_month', 'column_4_day_of_week', 'column_4_hour'] +
+                                                    text_columns)
+    assert input_feature_names['Elastic Net'] == (['column_3', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
+                                                   'column_2_1', 'column_2_2', 'column_2_3', 'column_2_4', 'column_2_5', 'column_2_6',
+                                                   'column_4_year', 'column_4_month', 'column_4_day_of_week', 'column_4_hour'] +
+                                                  text_columns)
     assert input_feature_names['Logistic Regression'] == ['Random Forest', 'Elastic Net']
 
 
+def test_component_graph_types_multi_input_1():
+    graph = {'Select numeric col_2': [SelectColumns],
+             'Imputer numeric col_2': [Imputer, 'Select numeric col_2.x'],
+             'Scaler col_2': [StandardScaler, 'Imputer numeric col_2.x'],
+             'Select categorical col_1': [SelectColumns],
+             'Imputer categorical col_1': [Imputer, 'Select categorical col_1.x'],
+             'OneHot col_1': [OneHotEncoder, 'Imputer categorical col_1.x'],
+             'Pass through col_3': [SelectColumns],
+             'Random Forest': [RandomForestClassifier, 'Scaler col_2.x', 'OneHot col_1.x', 'Pass through col_3.x']}
+
+    X = pd.DataFrame({'column_1': ['a', 'b', 'c', 'd', 'a', 'a', 'b', 'c', 'b'],
+                      'column_2': [1, 2, 3, 4, 5, 6, 5, 4, 3],
+                      'column_3': [True, False, True, False, True, False, True, False, False]})
+    y = pd.Series([1, 0, 1, 0, 1, 1, 0, 0, 0])
+    # woodwork would infer this as boolean by default -- convert to a numeric type
+    X = infer_feature_types(X, {"column_3": "integer"})
+
+    component_graph = ComponentGraph(graph)
+    # we don't have feature type selectors defined yet, so in order for the above graph to work we have to
+    # specify the types to select here.
+    # if the user-specified woodwork types are being respected, we should see the pass-through column_3 staying as numeric,
+    # meaning it won't cause a modeling error when it reaches the estimator
+    component_graph.instantiate({'Select numeric col_2': {'columns': ['column_2']},
+                                 'Select categorical col_1': {'columns': ['column_1']},
+                                 'Pass through col_3': {'columns': ['column_3']}})
+    assert component_graph.input_feature_names == {}
+    component_graph.fit(X, y)
```

Comment:
So is this a smoke test for the estimator when fit() is called?  Where is the comment-mentioned test for column_3?

Reply:
Lol yeah this is a little confusing. The assert a couple lines down checks that column_3 was provided as an input to the RF estimator. Before the impl changes in this PR, this test would fail during `fit`, because the code where we concatenate the various inputs would fall over.

I'll see if I can find a clean way to have this test focus on the types which each component sees, rather than on the feature names. That would definitely make the value of this test easier to follow.

Reply:
Ok I think I got it